src/Pos/ReportServer/Server.hs

```diff
@@ -48,31 +55,62 @@ limitBodySize limit application request responseHandler =
 
 liftAndCatchIO
     :: (MonadIO m, MonadCatch m, MonadBaseControl IO m)
-    => IO a -> S.ActionT (ReportServerException) m a
-liftAndCatchIO action = either raise pure =<< tryAll (liftIO action)
+    => IO a -> m (Either ReportServerException a)
+liftAndCatchIO = tryAll . liftIO
 
-reportServerApp :: LogsHolder -> S.ScottyT ReportServerException IO ()
-reportServerApp holder = do
-    post "/report" $ do
-        (payload :: ReportInfo) <-
-            either failPayload pure . eitherDecode =<< param "payload"
-        logFiles <-
-            map (bimap LT.toStrict $ TE.decodeUtf8 . BSL.toStrict . fileContent) <$>
-            files
-        let missingLogs = rLogs payload \\ map fst logFiles
-        unless (null missingLogs) $ failMissingLogs missingLogs
-        let neededLogs = filter ((`elem` rLogs payload) . fst) logFiles
-        let payloadFile = ("payload.json", prettifyJson payload)
-        cInfo <- getClientInfo
-        let clientInfoFile = ("client.info", prettifyJson cInfo)
-        liftAndCatchIO $ addEntry holder $ payloadFile : clientInfoFile : neededLogs
-        status status200
-    notFound err404
+withStatus :: Status -> T.Text -> Request -> Response
+withStatus status msg req = responseLBS status (requestHeaders req) (encodeUtf8 msg)
+
+-- | Gets the list of the uploaded files.
+bodyParse :: Request -> IO ([Param], [File LByteString])
+bodyParse request = do
+    let parseBodyOptions = defaultParseRequestBodyOptions
+    parseRequestBodyEx parseBodyOptions lbsBackEnd request
+
+-- Tries to retrieve the `ReportInfo` from the raw `Request`.
+param :: ByteString -> [Param] -> ByteString
+param key = fromMaybe mempty . lookup key
```

Comment:
No particular reason, I just though that it would have been nice to keep this function pure, as we will fail (with an exception) a couple of lines below during `eitherDecodeStrict`. Not particularly fussed though, I'm happy to restore the old behaviour!